pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -63,7 +63,15 @@ func (in *NodeClaimSpec) convertTo(ctx context.Context, v1beta1nc *v1beta1.NodeC
 		}
 	}
 
-	// Need to implement Kubelet Conversion
+	if kubeletAnnotation != "" {
+		v1beta1kubelet := &v1beta1.KubeletConfiguration{}
+		err := json.Unmarshal([]byte(kubeletAnnotation), v1beta1kubelet)
+		if err != nil {
+			return err
```

Comment:
add a wrapper so we can know where this is coming from
```suggestion
			return fmt.Errorf("unmarshaling kubelet config annotation, %w", err)
```